src/inference/src/model_reader.cpp

```diff
@@ -15,6 +14,20 @@
 #include "transformations/utils/utils.hpp"
 
 namespace {
+ov::element::Type to_legacy_type(const ov::element::Type& legacy_type, bool input) {
```

Comment:
I think will be cleaner if there will be two functions one for input and one when not use input. Then bool flag input want be required (is compile time value now) and logic each of it will be simpler. When runtime value for input flag has to handled then 3rd function can be added which just be switch to select correct implementation.

Reply:
I don't want to change logic in this PR. Otherwise we can bring bugs which are hard to locate